snmp/datadog_checks/snmp/snmp.py

```diff
@@ -531,19 +546,12 @@ def submit_metric(self, name, snmp_value, forced_type, tags, options):
             metric_name = self.normalize(name, prefix='snmp')
 
         if forced_type is not None:
-            try:
-                metric = as_metric_with_forced_type(snmp_value, forced_type, options)
-            except Exception as e:
-                self.log.error('Unable to coerce %s to %s: %s', name, forced_type, e)
-                return
-            if metric is None:
-                raise ConfigurationError('Invalid forced-type {!r} for metric {!r}'.format(forced_type, name))
+            metric = as_metric_with_forced_type(snmp_value, forced_type, options)
```

Comment:
IMO we shouldn't fail the whole check using `ConfigurationError` in `submit_metric`. We want the check to continue process the rest of the metrics even if there is an issue with one of them.